doc/en/components/grids/_shared/row-pinning.md

```diff
@@ -241,45 +236,60 @@ igRegisterScript("WebGridRowPinCellTemplate", (ctx) => {
 </{ComponentSelector}>
 ```
 
+On click of the custom icon the pin state of the related row can be changed using the row's API methods.
+
+```typescript
+public togglePinning(row: IgxGridRow, event) {
+    event.preventDefault();
+    if (row.pinned) {
+        row.unpin();
+    } else {
+        row.pin();
+    }
+}
+```
+<!-- end: Angular -->
+
 <!-- WebComponents -->
 ```html
-<{ComponentSelector} id="grid" primary-key="ID" auto-generate="false">
-    <igc-column id="column" width="70px">
+<{ComponentSelector} id="grid" primary-key]="ID" auto-generate="false">
+    <igc-column id="column1" name="column1">
     </igc-column>
 </{ComponentSelector}>
 ```
-```ts
+
+```typescript
 constructor() {
-    var grid = document.getElementById('grid') as IgcGridComponent;
-    var column = document.getElementById('column') as IgcColumnComponent;
-    grid.data = this.data;
-    column.bodyTemplate = this.cellPinCellTemplate;
+    var grid = this.grid = document.getElementById('grid') as {ComponentName};
```

Comment:
{ComponentName}Component

Reply:
Also there's no need for `this.grid`, `this.column` to be set. They aren't used anywhere.